File: frontend/src/api/adminService.js
import api from "./axiosConfig";

// ---------- Module 1 - Dashboard ----------

export const getDashboard = async () => {
  const res = await api.get("/admin/dashboard");
  return res.data;
};

// ---------- Module 2 - User Management ----------

export const getUsers = async (search, role) => {
  const params = {};
  if (search) params.search = search;
  if (role) params.role = role;
  const res = await api.get("/admin/users", { params });
  return res.data;
};

export const getUserById = async (id) => {
  const res = await api.get(`/admin/users/${id}`);
  return res.data;
};

export const suspendUser = async (id) => {
  const res = await api.patch(`/admin/users/${id}/suspend`);
  return res.data;
};

export const activateUser = async (id) => {
  const res = await api.patch(`/admin/users/${id}/activate`);
  return res.data;
};

// ---------- Module 3 - Vehicle Verification ----------

export const getPendingVehicles = async () => {
  const res = await api.get("/admin/vehicles/pending");
  return res.data;
};

export const approveVehicle = async (id) => {
  const res = await api.patch(`/admin/vehicles/${id}/approve`);
  return res.data;
};

export const rejectVehicle = async (id, reason) => {
  const res = await api.patch(`/admin/vehicles/${id}/reject`, { reason });
  return res.data;
};

// ---------- Module 4 - Ride Monitoring ----------

export const getRides = async (status) => {
  const params = {};
  if (status) params.status = status;
  const res = await api.get("/admin/rides", { params });
  return res.data;
};

export const getRideById = async (id) => {
  const res = await api.get(`/admin/rides/${id}`);
  return res.data;
};

// ---------- Module 5 - Booking Monitoring ----------

export const getBookings = async (status) => {
  const params = {};
  if (status) params.status = status;
  const res = await api.get("/admin/bookings", { params });
  return res.data;
};

export const getBookingById = async (id) => {
  const res = await api.get(`/admin/bookings/${id}`);
  return res.data;
};

File: backend/src/main/java/com/ecolift/service/AdminService.java
package com.ecolift.service;

import com.ecolift.dto.response.AdminDashboardResponse;
import com.ecolift.dto.response.AdminRideResponse;
import com.ecolift.dto.response.AdminUserResponse;
import com.ecolift.dto.response.BookingResponse;
import com.ecolift.dto.response.VehicleResponse;

import java.util.List;

/**
 * Admin Management Module service. The Admin does not participate in rides;
 * this service only aggregates/reuses the existing User/Vehicle/Ride/Booking
 * services for platform management, verification, and monitoring.
 */
public interface AdminService {

    // Module 1 - Dashboard
    AdminDashboardResponse getDashboard();

    // Module 2 - User Management
    List<AdminUserResponse> getUsers(String search, String role);
    AdminUserResponse getUserById(Long userId);
    void suspendUser(Long userId);
    void activateUser(Long userId);

    // Module 3 - Vehicle Verification
    List<VehicleResponse> getPendingVehicles();
    VehicleResponse approveVehicle(Long vehicleId);
    VehicleResponse rejectVehicle(Long vehicleId, String reason);

    // Module 4 - Ride Monitoring
    List<AdminRideResponse> getRides(String status);
    AdminRideResponse getRideById(Long rideId);

    // Module 5 - Booking Monitoring
    List<BookingResponse> getBookings(String status);
    BookingResponse getBookingById(Long bookingId);
}
